drivers/builtin/src/nist_kw.c

```diff
@@ -206,11 +206,8 @@ psa_status_t mbedtls_nist_kw_wrap(mbedtls_svc_key_id_t key,
                 R2 = output + KW_SEMIBLOCK_LENGTH;
             }
         }
-        ret = psa_cipher_finish(&wrap_operation,
-                                outbuff + olen,
-                                sizeof(outbuff) - olen,
-                                &part_length);
-        if (ret != PSA_SUCCESS) {
+        if (olen != 16) {
+            ret = PSA_ERROR_BAD_STATE;
```

Comment:
`BAD_STATE` doesn't really make sense here since there's no context that's left in a bad state. But since this is a “shouldn't happen” error, it doesn't really matter. Technically `CORRUPTION_DETECTED` would be the most fitting error.